src/renderers/dom/shared/__tests__/ReactBrowserEventEmitter-test.js

```diff
@@ -435,4 +435,54 @@ describe('ReactBrowserEventEmitter', () => {
     expect(idCallOrder[1] === PARENT).toBe(true);
     expect(idCallOrder[2] === GRANDPARENT).toBe(true);
   });
+
+  describe('local listener attachment', function() {
+    it('does not attach a local listener a second time on update', () => {
+      var container = document.createElement('div');
+      var spy = jest.fn();
+
+      class TestComponent extends React.Component {
+        render() {
+          return <div onTouchMove={spy}>Test</div>;
+        }
+      }
+
+      var component = ReactDOM.render(<TestComponent />, container);
+      var el = container.querySelector('div');
+
+      // Force an update to cause a re-render
+      component.forceUpdate();
```

Comment:
This is not enough to cause the ensureListeningTo to be called twice because it will bail out on function object equality here. https://github.com/facebook/react/blob/master/src/renderers/dom/fiber/ReactDOMFiberComponent.js#L672

You need to change the callback instance. For example by doing `onTouchMove={() => spy()}`.